aten/src/ATen/native/ReduceOps.cpp

```diff
@@ -599,6 +599,98 @@ Tensor& diff_out(const Tensor& self, int64_t n, int64_t dim, const c10::optional
   }
 }
 
+void axis_check(IntArrayRef& axis, int64_t bound) {
+  std::unordered_set<int64_t> dimensions;
+  for (const auto i : c10::irange(axis.size())) {
+    int64_t pos_axis  = axis[i] >= 0 ? axis[i] : axis.size()+axis[i];
+    TORCH_CHECK( dimensions.find(pos_axis) == dimensions.end(), "torch.gradient requires axis to be unique");
+    TORCH_CHECK( (axis[i] < bound) && (axis[i] >= (-1*bound)), "torch.gradient's axis " + std::to_string(axis[i]) +" is out of bounds for array of dimension " + std::to_string(bound)  );
+    dimensions.insert( pos_axis);
+  }
+}
+
+std::vector<Tensor> gradient(const Tensor& self,
+                             TensorList spacing,
+                             IntArrayRef axis,
+                             int64_t edge_order) {
+  axis_check(axis, self.dim());
+  TORCH_CHECK(self.scalar_type() != ScalarType::Byte, "torch.gradient does not support uint8 as input.");
+  TORCH_CHECK( spacing.size() == axis.size(), "torch.gradient should receive axis and spacing size to be equal.");
```

Comment:
This is not quite accurate. It's only if axis is specified that the number of elements in spacing should equal the number of (unique) axes.

Reply:
if axis is not provided, then corresponding gradient functions are given in 661 and 690, which are calling this function after creating corresponding axis vector. That means, when we are called this function we are positive that there are spacing and axis elements provided (i.e. non-empty)